internal/services/v1alpha1/lookupwatch.go

```diff
@@ -0,0 +1,204 @@
+			Unary: grpcmw.ChainUnaryServer(grpcutil.DefaultUnaryMiddleware...),
+		},
+		permissionsClient: permissionsClient,
+		watchClient:       watchClient,
+	}
+}
+
+func (lws *lookupWatchServiceServer) WatchAccessibleResources(req *v1alpha1.WatchAccessibleResourcesRequest, resp v1alpha1.LookupWatchService_WatchAccessibleResourcesServer) error {
+
+	ctx := resp.Context()
+
+	done := make(chan bool)
+	errCh := make(chan error, 1)
+
+	go func() {
+
+		// start a watch over all changes
+		stream, err := lws.watchClient.Watch(ctx, &v1.WatchRequest{
+			OptionalStartCursor: req.GetOptionalStartCursor(),
+		})
+		if err != nil {
+			errCh <- fmt.Errorf("todo(jon-whit): come up with a better error")
+			return
+		}
+
+		for {
+			watchResponse, err := stream.Recv()
+			if err == io.EOF {
+				done <- true
+				return
+			}
+			if err != nil {
+				errCh <- fmt.Errorf("todo(jon-whit): come up with a better error")
+				return
+			}
+
+			updates := watchResponse.GetUpdates()
+			revision := watchResponse.GetChangesThrough()
+
+			permissionUpdates := []*v1alpha1.PermissionUpdate{}
+
+			maxConcurrentUpdates := 10
+			limiter := make(chan struct{}, maxConcurrentUpdates)
+
+			for _, update := range updates {
+
+				operation := update.GetOperation()
+				if operation == v1.RelationshipUpdate_OPERATION_UNSPECIFIED {
+					continue
+				}
+
+				limiter <- struct{}{}
+
+				go func(update *v1.RelationshipUpdate) {
+
+					// todo(jon-whit): figure out how to do this in a more memory efficient manner
+
+					resources := []*v1.ObjectReference{}
+					_ = resources
+					subjects := []*v1.SubjectReference{}
+					_ = subjects
+
+					go func() {
+						// resourceStream, err := v1alpha1.ReachableResources(context.TODO(), &v1alpha1.ReachResourcesRequest{
+						// 	StartingResource: update.GetRelationship().GetResource(),
+						// 	TargetObjectType: req.GetResourceObjectType(),
+						// })
+						// if err != nil {
+						// 	// handle error
+						// }
+
+						// for {
+						// 	resp, err := resourceStream.Recv()
+						// 	if err == io.EOF {
+						// 		return
+						// 	}
+						// 	if err != nil {
+						// 		// handle error
+						// 	}
+
+						// 	resources = append(resources, resp.GetFoundResource())
+						// }
+					}()
+
+					go func() {
+						// subjectStream, err := v1alpha1.LookupSubjects(context.TODO(), &v1alpha1.LookupSubjectsRequest{
+						// 	Resource:          update.GetRelationship().GetResource(),
+						// 	TargetSubjectType: req.GetSubjectObjectType(),
+						// })
+						// if err != nil {
+						// 	// handle error
+						// }
+
+						// for {
+						// 	resp, err := subjectStream.Recv()
+						// 	if err == io.EOF {
+						// 		return
+						// 	}
+						// 	if err != nil {
+						// 		// handle error
+						// 	}
+
+						// 	subjects = append(subjects, resp.GetFoundRelationship().GetSubject())
+						// }
+					}()
+
+					for _, resource := range resources {
```

Comment:
We could evaluate these checks concurrently. For example, something like:

```go
var wg sync.WaitGroup
for _, resource := range resources {
    wg.Add(1)
    go func() {
        defer wg.Done()

        for _, subject := range subjects {
            wg.Add(1)
            go func() {
                defer wg.Done()

                Check(resource, subject, ...)
            }()
        }
    }()
}

wg.Wait()
```

My only concern there is just the sheer number of concurrent requests that might be issued. We could throttle it like I did with the `limiter` approach above, but I wanted to get your guys' opinion.